- [x] specify neither ARN nor IAM Role Name

```
ec2c launch \
       --ami $AMI_ID \
       --publicip \
       --az $AVAILABILITY_ZONE \
       --type $INSTANCE_TYPE \
       --key $SSH_KEY_NAME \
       --sg $SECURITY_GROUPS \
       --subnet $SUBNET_ID \
       --userData ./user-data \
       --volumeSize $VOLUME_SIZE \

===> Successfully launched! Instance ID: i-
```

ok,create instance.
- [x] specify ARN only

```
ec2c launch \
       --ami $AMI_ID \
       --publicip \
       --az $AVAILABILITY_ZONE \
       --type $INSTANCE_TYPE \
       --key $SSH_KEY_NAME \
       --sg $SECURITY_GROUPS \
       --subnet $SUBNET_ID \
       --userData ./user-data \
       --volumeSize $VOLUME_SIZE \
       --arn $ARN \
===> Successfully launched! Instance ID: i-
```

ok, create instance.
- [x] specify IAM Role Name only

```
ec2c launch \
       --ami $AMI_ID \
       --publicip \
       --az $AVAILABILITY_ZONE \
       --type $INSTANCE_TYPE \
       --key $SSH_KEY_NAME \
       --sg $SECURITY_GROUPS \
       --subnet $SUBNET_ID \
       --userData ./user-data \
       --volumeSize $VOLUME_SIZE \
       --roleName $ROLE_NAME \
===> Successfully launched! Instance ID: i-
```

ok. create instance.
- [x] specify both ARN and IAM Role Name => Not Supported

```
ec2c launch \
       --ami $AMI_ID \
       --publicip \
       --az $AVAILABILITY_ZONE \
       --type $INSTANCE_TYPE \
       --key $SSH_KEY_NAME \
       --sg $SECURITY_GROUPS \
       --subnet $SUBNET_ID \
       --userData ./user-data \
       --volumeSize $VOLUME_SIZE \
       --roleName $ROLE_NAME \
       --arn $ARN \
```

```
panic: InvalidParameterCombination: The parameter 'iamInstanceProfile.name' may not be used in combination with 'iamInstanceProfile.arn'
    status code: 400, request id: fdd5a2c1-8498-439a-a4d9-78927e04b813
```

I added validate check.

```
InvalidParameterCombination: The parameter 'roleName' and 'arn' can not be used in combination
```

plz review 🏹

--- a/command/launch.go
+++ b/command/launch.go
@@ -56,7 +56,7 @@ func (c *LaunchCommand) Run(args []string) int {
 	flags.StringVar(&userData, "userData", "", "User data")
 	flags.Int64Var(&volumeSize, "volumeSize", 8, "Volume size (default: 8)")
 	flags.StringVar(&volumeType, "volumeType", "gp2", "Volume type (default: gp2)")
-	flags.StringVar(&arn, "arn", "", "Amazon Resource Name")
+	flags.StringVar(&arn, "arn", "", "IAM Instance Profile ARN")
 	flags.StringVar(&roleName, "roleName", "", "IAM Role Name")
 
 	if err := flags.Parse(args[0:]); err != nil {

--- a/command/launch.go
+++ b/command/launch.go
@@ -56,7 +56,7 @@ func (c *LaunchCommand) Run(args []string) int {
 	flags.StringVar(&userData, "userData", "", "User data")
 	flags.Int64Var(&volumeSize, "volumeSize", 8, "Volume size (default: 8)")
 	flags.StringVar(&volumeType, "volumeType", "gp2", "Volume type (default: gp2)")
-	flags.StringVar(&arn, "arn", "", "ARN")
+	flags.StringVar(&arn, "arn", "", "Amazon Resource Name")
 	flags.StringVar(&roleName, "roleName", "", "IAM Role Name")
 
 	if err := flags.Parse(args[0:]); err != nil {

--- a/command/launch.go
+++ b/command/launch.go
@@ -5,7 +5,6 @@ import (
 	"flag"
 	"fmt"
 	"io/ioutil"
-	"os"
 	"strings"
 
 	"github.com/aws/aws-sdk-go/aws"
@@ -67,11 +66,6 @@ func (c *LaunchCommand) Run(args []string) int {
 	for 0 < flags.NArg() {
 		arguments = append(arguments, flags.Arg(0))
 		flags.Parse(flags.Args()[1:])
-	}
-
-	if arn != "" && roleName != "" {
-		fmt.Fprintln(os.Stderr, "InvalidParameterCombination: The parameter 'roleName' and 'arn' can not be used in combination")
-		return 1
 	}
 
 	for _, id := range strings.Split(securityGroupIDs, ",") {

